lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +178,37 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        url = article_bs.get('href')
+        if isinstance(url, str) and url.startswith('https://www.volga-tv.ru/news/'):
+            return url
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self.config.get_seed_urls():
+            response = make_request(seed_url, self.config)
+            main_bs = BeautifulSoup(response.text, 'lxml')
+            all_links = main_bs.find_all('a')
+            for link in all_links:
+                url = self._extract_url(link)
+                if url.startswith("/news"):
+                    new_url = 'https://www.volga-tv.ru' + url
+                    if new_url not in self.urls and len(self.urls) < self.config.get_num_articles():
```

Comment:
consider breaking the cycle when the required number of articles is collected, it may save some time

Reply:
done

Reply:
As of now, you only break the smaller cycle over the articles collected from one of the seed URLs. I believe that once you've collected the required number of articles, there is no need to iterate over the remaining seed URLs anymore.